### PR TITLE
ENH: Add utf-8 support for savetxt

### PR DESCRIPTION
**Pull request by DavidToca**
Add support for utf-8 strings for **savetxt**

related issues
  - https://github.com/numpy/numpy/issues/4939 
  - https://github.com/numpy/numpy/issues/4543

### CODE REVIEW

File: numpy/compat/py3k.py [hunk 1]
```diff
@@ -26,17 +26,17 @@
 
     def asunicode(s):
         if isinstance(s, bytes):
-            return s.decode('latin1')
```
Comment (pv) [inline numpy/compat/py3k.py]: These should not be touched. Any changes should be restricted to npyio.py.

**Comment by pv** [comment]
I don't think hardcoding an encoding like this is the correct way forward.

There are a number of things to address in savetxt/loadtxt, and the strategy should be explicitly discussed to be sure to get it right:

- The savetxt handling of byte string formatting is buggy on Py3 --- this traces back to `%` formatting being not supported for bytes on Python < 3.5.
- Writing to text streams vs. byte streams.
- Selecting the encoding to use for cases where conversions are necessary.

You also need to add corresponding tests to the test suite for a PR to be accepted.

**Comment by DavidToca** [comment]
@pv thanks for your reviewing, so, should I add this as an issue, or post it in on the dev mailing list in order to be discussed?

**Comment by pv** [comment]
What I mean by "explicitly discussed" is that I think it would be useful to first write down an overview/spec of how you think it should work, for example in the pull request description above. For example, what should happen with dtype='S and type='U' strings etc., how to select an encoding, whether there should be a possibility to write/read text streams instead of just byte streams --- once there's such an overview/spec of the intended behavior, it's easier to check whether the code does what was intended.

**Comment by juliantaylor** [comment]
savetxt has an encoding option in 1.14 so this PR can be closed

**Pull request closed without merging**